examples/simple_intersection/example_intersection.cpp

```diff
@@ -0,0 +1,127 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+// Do intersection queries using the same objects for the queries as the objects
+// used in BVH construction that are located on a regular spaced
+// three-dimensional grid.
```

Comment:
If you do self-intersection, the grid brings nothing to the table but extra complexity

Reply:
In my opinion, it's useful to have a complete example that users can easily relate to it; even if that means more code than absolutely necessary.